File: bc2026/src/bestsofar/RobotPlayer.cpp
#include "RobotPlayer.hpp"

#include <iostream>

#include "Baby.hpp"
#include "King.hpp"
#include "Utils/Globals.hpp"

void RobotPlayer::run(RobotController rc) {
    Utils::Globals::init(rc);

    while (true) {
        try {
            Utils::Globals::startTurn();

            switch (rc.getType()) {
                case UnitType::BABY_RAT:
                    Baby::run();
                    break;

                case UnitType::RAT_KING:
                    King::run();
                    break;

                default:
                    break;
            }

            Utils::Globals::endTurn();

            Clock::yield();
        } catch (const GameActionException& e) {
            std::cout << "GameActionException" << '\n';
            std::cout << e.what() << '\n';
        } catch (const std::exception& e) {
            std::cout << "Exception" << '\n';
            std::cout << e.what() << '\n';
        }
    }
}

File: py/bc2026/src/bestsofar/RobotPlayer.py
# Port of bestsofar/RobotPlayer.java. See py/CONVENTIONS.md.
import traceback

from api import *
import Baby
import King
from Utils.Globals import Globals


class RobotPlayer:
    @staticmethod
    def run(rc):
        Globals.init(rc)

        while True:
            try:

                Globals.startTurn()

                # Java: switch on getType() - the expression is evaluated once
                t = rc.getType()
                if t == UnitType.BABY_RAT:
                    Baby.Baby.run()

                elif t == UnitType.RAT_KING:
                    King.King.run()

                Globals.endTurn()


                Clock.doYield()
            except GameActionException:
                print("GameActionException")
                traceback.print_exc()

            except Exception:
                print("Exception")
                traceback.print_exc()
